src/main/java/org/cyclops/integrateddynamics/core/evaluate/variable/ValueTypeListProxyBase.java

```diff
@@ -77,6 +80,15 @@ public boolean equals(Object obj) {
         return Arrays.equals(o, o2);
     }
 
+    @Override
+    public int hashCode() {
+        int result = 1;
+        result = 37 * result + getName().hashCode();
+        result = 37 * result + getValueType().hashCode();
+        result = 37 * result + (!this.isInfinite() ? Arrays.hashCode(Iterables.toArray(this, Object.class)) : 0);
```

Comment:
I think `Arrays.hashCode(Iterables.toArray(...))` can be made faster by manually calculating the hashcode using a simple foreach loop. That way we avoid allocating a new array.

Reply:
Done.